sklearn/linear_model/base.py

```diff
@@ -273,6 +273,11 @@ def _set_intercept(self, X_offset, y_offset, X_scale):
         """Set the intercept_
         """
         if self.fit_intercept:
+            if isinstance(self.coef_, np.ndarray):
```

Comment:
Since error happened at `dtype = self.coef_.dtype`, with the message:

```
AttributeError: 'list' object has no attribute 'dtype'
```

I think `self.coef_` is not always a numpy array?